chain/chain/src/chain.rs

```diff
@@ -1333,11 +1334,17 @@ impl Chain {
             on_challenge,
         );
 
-        if res.is_ok() {
-            metrics::BLOCK_PROCESSED_TOTAL.inc();
-            success_timer.stop_and_record();
-        } else {
-            success_timer.stop_and_discard();
+        match &res {
+            Ok(_) => {
+                metrics::BLOCK_PROCESSED_TOTAL.inc();
+                success_timer.stop_and_record();
+            }
+            Err(_) => {
+                if let Err(e) = self.save_block_height_processed(block_height) {
```

Comment:
can you add a comment explaining why we still want to do something during the original error processing?

Reply:
seems that this method is only used here.

And from what I see: we use the 'height requested' column only to 'reject' incoming blocks, that we didn't explicitly request.

(if there is a new incoming block and we already have height processed 'marked as true' - then we'll not even attempt to process it).

that's why we probably mark the height as processed, even if the block_processing failed.
You might want to mention some of the things above in the comment here.